autopush/db.py

```diff
@@ -306,8 +314,13 @@ def delete_messages_for_channel(self, uaid, channel_id):
     @track_provisioned
     def fetch_messages(self, uaid, limit=10):
         """Fetches messages for a uaid"""
-        return self.table.query_2(uaid__eq=uaid, chidmessageid__gt=" ",
-                                  consistent=True, limit=limit)
+        # Eagerly fetches all results in the result set.
+        messages = list(self.table.query_2(uaid__eq=uaid,
+                        chidmessageid__gt=" ", consistent=True, limit=limit))
+        for message in messages:
+            message["data"] = urlsafe_b64encode(message["data"].value)
+            message.mark_clean()
```

Comment:
Why does it need to be marked clean? Do we call save on it somewhere? (It shouldn't auto-save these.... does it?)